### ISSUE TITLE
variable substitution in project pre/post/etc build step....

### ISSUE THREAD

**Issue by Brane2** [eranif/codelite#1060]
I have Codelite 9.0 compiled with gcc-5.2.0 on latest, mostly stable 64.bit Gentoo system.

For the life of me, I can't make Codelite substitute automatic variables in "Pre build" step as listed under "Help" option. For example, line:

echo "ProjectPath is $(ProjectPath)"
echo "WorkspacePath is $(WorkspacePath)"
...

returns when building:

./PCB.sh: line 5: ProjectPath: command not found
ProjectPath is
./PCB.sh: line 6: WorkspacePath: command not found
WorkspacePath is

IOW, both variables expand to empty string.
Every other of "Available Macros" that I took the time to try, behave the same - I get empty string.
What gives ?

BTW: While at it, could you please check various other places, like Project Settings/General/{Program Arguments , Working Directory, etc} for which other users have reported practically identical issues ?
I have currently no simple way to test these, but i strongly suspect it is the same bug.

**Comment by eranif** [comment]
> BTW: While at it, could you please check various other places, like Project Settings/General/{Program 
> Arguments , Working Directory, etc} for which other users have reported practically identical issues ?
> I have currently no simple way to test these, but i strongly suspect it is the same bug.

I am not aware of any bugs in other locations, if you do find one, please open a separate bug
I can only fix those I see

**Comment by MoutonSanglant** [comment]
I have the same issue in Codelite 10.0, compiled with gcc-7.1.1 on Arch Linux 64-bit.

```
echo "Workspace path: $(WorkspacePath)"
```
results in
```
./example.sh: line 4: WorkspacePath: command not found
Workspace path:
```

As a workaround, I use environnment variables (set per workspace), but I would prefer to use predefined macros.

```
workspace_path=$(WorkspacePath)
```

**Comment by eranif** [comment]
I just tested this and it seems to work perfectly using the latest build of CodeLite.

> ./example.sh: line 4: WorkspacePath: command not found

Where did `example.sh` came from? its not something that CodeLite creates...

**Comment by MoutonSanglant** [comment]
It is the project's name (juste like `PCB.sh`, in Brane2's message, PCB is certainly hist project's name).

I will try the latest code during this week and give you my feedback.

**Comment by MoutonSanglant** [comment]
I just built latest (10.0.7) and bug is still present so I investigated a bit more and it only happens when `Enable custom build` is checked in the project's settings. Whenever the option is activated, a shell script is created (inside the specified `WorkingDirectory`) after the user launch a build. Output of `env` in post/pre build script shows that only the variables set in `Workspace Settings` are accessible.

I attached a sample project with buggy configuration.

[PrePostBuildTest.zip](https://github.com/eranif/codelite/files/1276545/PrePostBuildTest.zip)
